src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Subscriptions.cs

```diff
@@ -0,0 +1,272 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Maestro.ScenarioTests.ObjectHelpers;
+using Microsoft.DotNet.Darc;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    [Category("ScenarioTest")]
+    public class ScenarioTests_Subscriptions : MaestroScenarioTestBase
+    {
+        private TestParameters _parameters;
+
+        private string repo1Name = "maestro-test1";
+        private string repo2Name = "maestro-test2";
+        private string repo3Name = "marstro-test3";
+        private string channel1Name = "subscriptionTestChannel1";
+        private string channel2Name = "subscriptionTestChannel2";
+
+        [SetUp]
+        public async Task InitializeAsync()
+        {
+            _parameters = await TestParameters.GetAsync();
+            SetTestParameters(_parameters);
+        }
+
+        [TearDown]
+        public Task DisposeAsync()
+        {
+            _parameters.Dispose();
+            return Task.CompletedTask;
+        }
+
+        [Test]
+        [Category("ScenarioTest")]
+        public async Task Subscriptions_EndToEnd()
+        {
+            TestContext.WriteLine("Subscription management tests...");
+
+            string repo1Uri = GetRepoUrl(repo1Name);
+            string repo2Uri = GetRepoUrl(repo2Name);
+            string repo3Uri = GetRepoUrl(repo3Name);
+            string repo1AzDoUri = GetAzDoRepoUrl(repo1Name);
+            string targetBranch = "master";
+
+            TestContext.WriteLine($"Creating channels {channel1Name} and {channel2Name}");
+            await using (AsyncDisposableValue<string> channel1 = await CreateTestChannelAsync(channel1Name).ConfigureAwait(false))
+            {
+                await using (AsyncDisposableValue<string> channel2 = await CreateTestChannelAsync(channel2Name).ConfigureAwait(false))
+                {
+
+                    TestContext.WriteLine("Testing various command line parameters of add-subscription");
+                    await using AsyncDisposableValue<string> subscription1Id = await CreateSubscriptionAsync(
+                        channel1Name, repo1Name, repo2Name, targetBranch, "everyWeek", "maestro-auth-test");
+
+                    Subscription expectedSubscription1 = SubscriptionBuilder.BuildSubscription(
+                        repo1Uri, 
+                        repo2Uri, 
+                        targetBranch, 
+                        channel1Name, 
+                        subscription1Id.Value, 
+                        UpdateFrequency.EveryWeek, 
+                        false, 
+                        null, 
+                        null);
+
+                    string expectedSubscription1Info = UxHelpers.GetTextSubscriptionDescription(expectedSubscription1, null);
```

Comment:
[nit] null is the default for the merge policies param, so I don't think it's worth specifying it here and in other places where this method is called with a null param.